Scripts/RemediationScripts/Remediate-DoNotUseManagementCertificates.ps1

```diff
@@ -0,0 +1,526 @@
+
+    # Determine if the current user has the required permissions to run this script.
+    $canRunScript = $($($classicAdmins | Measure-Object).Count -gt 0)
+
+    if (-not $canRunScript)
+    {
+        Write-Host "This script can only be run by one of: $($eligibleRoles -join ", ")" -ForegroundColor $([Constants]::MessageType.Error)
+        break
+    }
+    else
+    {
+        Write-Host "$($context.Account.Id) has the required permissions." -ForegroundColor $([Constants]::MessageType.Update)
+    }
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 2 of 4] Preparing to fetch all management certificates..."
+
+    $certificates = @()
+
+    # No file path provided as input to the script. Fetch all management certificates in the Subscription.
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {
+        Write-Host "Fetching all management certificates for Subscription: $($context.Subscription.SubscriptionId)"
+        $certificateDetails = [XML]$managementCertificate.GetBySubscriptionId($context.Subscription.SubscriptionId)
+        $certificates = $certificateDetails.GetElementsByTagName("SubscriptionCertificate") | ForEach-Object { $_ | Select-Object @{N='SubscriptionCertificatePublicKey';E={$_.SubscriptionCertificatePublicKey}},
+                                                                                                                                  @{N='SubscriptionCertificateThumbprint';E={$_.SubscriptionCertificateThumbprint}},
+                                                                                                                                  @{N='SubscriptionCertificateData';E={$_.SubscriptionCertificateData}},
+                                                                                                                                  @{N='Created';E={$_.Created}}
+                                                                                                             }
+    }
+    else
+    {
+        if (-not (Test-Path -Path $FilePath))
+        {
+            Write-Host "ERROR: Input file - $($FilePath) not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+
+        Write-Host "Fetching all management certificates from $($FilePath)"
+        $certificateDetails = Import-Csv -LiteralPath $FilePath
+        $certificates = $certificateDetails | Where-Object { ![String]::IsNullOrWhiteSpace($_.SubscriptionCertificateThumbprint) }
+    }
+
+    $totalCertificates = ($certificates | Measure-Object).Count
+
+    if ($totalCertificates -eq 0)
+    {
+        Write-Host "No management certificates found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+  
+    Write-Host "Found $($totalCertificates) certificate(s)." -ForegroundColor $([Constants]::MessageType.Update)
+
+    # Use the current directory to back up snapshots.
```

Comment:
Are we following same path in all scripts, if we keep always is same path that might be helpful for user to view history later.